ts/src/bitmex.ts

```diff
@@ -1857,14 +1860,33 @@ export default class bitmex extends Exchange {
             'ordType': orderType,
             'text': brokerId,
         };
-        if ((orderType === 'Stop') || (orderType === 'StopLimit') || (orderType === 'MarketIfTouched') || (orderType === 'LimitIfTouched')) {
-            const stopPrice = this.safeNumber2 (params, 'stopPx', 'stopPrice');
-            if (stopPrice === undefined) {
-                throw new ArgumentsRequired (this.id + ' createOrder() requires a stopPx or stopPrice parameter for the ' + orderType + ' order type');
-            } else {
-                request['stopPx'] = parseFloat (this.priceToPrecision (symbol, stopPrice));
-                params = this.omit (params, [ 'stopPx', 'stopPrice' ]);
+        const customTriggerType = (orderType === 'Stop') || (orderType === 'StopLimit') || (orderType === 'MarketIfTouched') || (orderType === 'LimitIfTouched');
+        // support for unified trigger format
+        const triggerPrice = this.safeNumberN (params, [ 'triggerPrice', 'stopPx', 'stopPrice' ]);
+        if ((triggerPrice !== undefined) && !customTriggerType) {
+            request['stopPx'] = parseFloat (this.priceToPrecision (symbol, triggerPrice));
+            const triggerDirection = this.safeString (params, 'triggerDirection');
+            params = this.omit (params, [ 'triggerPrice', 'stopPrice', 'stopPx', 'triggerDirection' ]);
+            const triggerAbove = (triggerDirection === 'above');
+            this.checkRequiredArgument ('createOrder', triggerDirection, 'triggerDirection', [ 'above', 'below' ]);
```

Comment:
@ttodua Question, if the user does not provide `triggerDirection` but provides the exchange-specific type ('StopLimit/LimitIfTouched/etc), shouldn't that be enough?

Reply:
nevermind just noticed that you already considered that

Reply:
will add a test for that though

Reply:
@ttodua when I try to place an order I always get 🤔 

```
ResponseBody:
 {"error":{"message":"Invalid stopPx for ordType","name":"HTTPError"}} 

InvalidOrder bitmex {"error":{"message":"Invalid stopPx for ordType","name":"HTTPError"}}
```

Reply:
Yeah we have a bug there, when we use the exchange-specific parameter we send `triggerPrice` instead of `stopPx`

```
{"symbol":"TRX_USDT","side":"Buy","orderQty":500000000,"ordType":"StopLimit","text":"CCXT","price":0.05,"triggerPrice":0.07} 
```